tutorial/creating_packages/configure_options_settings/conanfile.py

```diff
@@ -0,0 +1,71 @@
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout
+from conan.tools.scm import Git
+from conan.tools.build import check_min_cppstd
+
+
+class helloRecipe(ConanFile):
+    name = "hello"
+    version = "1.0"
+
+    # Optional metadata
+    license = "<Put the package license here>"
+    author = "<Put your name here> <And your email here>"
+    url = "<Package recipe repository url here, for issues about the package>"
+    description = "<Description of hello package here>"
+    topics = ("<Put some tag here>", "<here>", "<and here>")
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {"shared": [True, False], 
+               "fPIC": [True, False],
+               "with_fmt": [True, False]}
+
+    default_options = {"shared": False, 
+                       "fPIC": True,
+                       "with_fmt": True}
+
+    generators = "CMakeDeps"
+
+    def validate(self):
+        if self.info.options.with_fmt:
+            check_min_cppstd(self, "11")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        git = Git(self)
+        git.clone(url="https://github.com/conan-io/libhello.git", target=".")
+        git.checkout("optional_fmt")
```

Comment:
I don't love that we are using a moving target here in the example. Maybe at least use a clear comment, saying this is not good practice and that an immutable tag or commit should be used instead of a branch?

Reply:
Yes, I can release a version in the repo and use that. I'll change that.

Reply:
Finally, I'll add a comment so it's clear it's not a good practice, and from some point in the tutorial just use an specific commit